File: impl/src/org/ufomatics/archipelago/model/MetaMappingImpl.java
package org.ufomatics.archipelago.model;

import org.ufomatics.archipelago.model.MetaMapping;
import org.ufomatics.archipelago.model.MetaModel;
import org.ufomatics.archipelago.model.Database;

import java.util.Set;
import java.util.HashSet;
import java.io.Serializable;

/**
 * @author Jérôme Beau
 * @version 10 juin 2006 18:49:46
 */
public class MetaMappingImpl implements MetaMapping, Serializable {
    private MetaModel metaModel = new MetaModelImpl();
    private Set<Database> datasources = new HashSet<Database>();

    public MetaModel getMetaModel() {
        return metaModel;
    }

    public Set<Database> getDatasources() {
        return datasources;
    }

    public void setMetaModel(MetaModel metaModel) {
        this.metaModel = metaModel;
    }

    public void setDatasources(Set<Database> datasources) {
        this.datasources = datasources;
    }
}

File: src/archipelago/impl/model/MetaMappingImpl.ts
import {MetaMapping} from "archipelago/api/model/MetaMapping"
import {Database} from "archipelago/api/model/Database"
import {MetaModel} from "archipelago/api/model/MetaModel"
import {MetaModelImpl} from "archipelago/api/model/MetaModelImpl"
import {JSerializable} from "ts-jsdk"

/**
 */
export class MetaMappingImpl implements MetaMapping, JSerializable {

  private metaModel: MetaModel = new MetaModelImpl()
  private datasources = new Set<Database>()

  getMetaModel(): MetaModel {
    return this.metaModel
  }

  getDatasources(): Set<Database> {
    return this.datasources
  }

  setMetaModel(metaModel: MetaModel): void {
    this.metaModel = metaModel
  }

  setDatasources(datasources: Set<Database>): void {
    this.datasources = datasources
  }
}
